### PR TITLE
GANs in pl-examples updated for lightning-0.9

### PR DESCRIPTION
**Pull request by VasudevGupta7**
What does this PR do?

Updated GANs examples in pl-examples as per lightning version-0.9

## PR review
Anyone in the community is free to review the PR once the tests have passed.     

## Did you have fun?
I love coding.

### CODE REVIEW

File: pl_examples/domain_templates/generative_adversarial_net.py [hunk 1]
```diff
@@ -197,7 +198,7 @@ def main(args: Namespace) -> None:
     # If use distubuted training  PyTorch recommends to use DistributedDataParallel.
     # See: https://pytorch.org/docs/stable/nn.html#torch.nn.DataParallel
     trainer = Trainer()
-
+xw
```
Comment (Borda) [inline pl_examples/domain_templates/generative_adversarial_net.py:201]:
```suggestion

```

**Review by Borda — COMMENTED** [review]
can we add a test run, call the script with some minimal data as the other MNIST examples does...

File: pl_examples/domain_templates/generative_adversarial_net.py [hunk 2]
```diff
@@ -127,12 +128,12 @@ def training_step(self, batch, batch_idx, optimizer_idx):
             # adversarial loss is binary cross-entropy
             g_loss = self.adversarial_loss(self.discriminator(self(z)), valid)
             tqdm_dict = {'g_loss': g_loss}
-            output = OrderedDict({
-                'loss': g_loss,
-                'progress_bar': tqdm_dict,
-                'log': tqdm_dict
-            })
-            return output
+            result = TrainResult(
```
Comment (williamFalcon) [inline pl_examples/domain_templates/generative_adversarial_net.py:131]: i would also add checkpoint_on in this case since there is no val loop

Comment (VasudevGupta7) [reply pl_examples/domain_templates/generative_adversarial_net.py:131]: I will add that.

**Comment by Borda** [comment]
@VasudevGupta7 no need to close it, just implement the update in this PR

**Pull request closed without merging**